File: code/cryptography/src/morse_cipher/morse_code_generator.c
/* Part of Cosmos by OpenGenus Foundation */
//C implementation of morse code generator
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void 
encrypt(char* msg)
{
    int size = strlen(msg);
    char * morse_code = (char*) malloc(size * 5);

    // Morse Code for the numbers [0-10]
    char* num[] = {"-----", ".----", "..---", "...--", "....-", ".....", "-....", "--...", "---..", "----."};
    // Morse Code for the alphabet
    char* letters[] = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..",
                      "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};

    int i;
    for (i = 0;i < size; i++)
        if (msg[i] != ' ')
            if (msg[i] >= '0' && msg[i] <= '9')
                strcat(morse_code, num[msg[i] - 48]);
            else
                strcat(morse_code, letters[msg[i] - 65]);

        else
            strcat(morse_code, " ");

    printf("%s\n", morse_code);

    free(morse_code);
}

int 
main()
{
    char* msg = "CONVERT THIS MORSE CODE";
    encrypt(msg);

    return (0);
}

File: code/cryptography/src/morse_cipher/morse_code_generator.rb
# cryptography | morse cipher | morse code generator | RUBY
// Part of Cosmos by OpenGenus Foundation

=end


MORSE_CODES = {
  'a' => '.-', 'b' => '-...', 'c' => '-.-.',
  'd' => '-..', 'e' => '.', 'f' => '..-.',
  'g' => '--.', 'h' => '....', 'i' => '..',
  'j' => '.---', 'k' => '-.-', 'l' => '.-..',
  'm' => '--', 'n' => '-.', 'o' => '---',
  'p' => '.--.', 'q' => '--.-', 'r' => '.-.',
  's' => '...', 't' => '-', 'u' => '..-',
  'v' => '...-', 'w' => '.--', 'x' => '-..-',
  'y' => '-.--', 'z' => '--..', ' ' => ' ',
  '1' => '.----', '2' => '..---', '3' => '...--',
  '4' => '....-', '5' => '.....', '6' => '-....',
  '7' => '--...', '8' => '---..', '9' => '----.',
  '0' => '-----'
}.freeze

def encode(string)
  text = string.downcase.split('')
  encoded_text = []

  text.each do |c|
    morse_char = MORSE_CODES[c]
    encoded_text.push(morse_char)
  end

  encoded_text.join('')
end

# TEST!
puts encode('Hello World') # => ......-...-..--- .-----.-..-..-..
